### PR TITLE
build: update nixpkgs and deps

### PR DESCRIPTION
**Pull request by tiagolobocastro**
test(bdd): update python packages
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    fix(resize): atomically check for the required size
    
    Ensures races don't lead into volume resize failures.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: specify patches for e2fsprogs
    
    This is required because the upstream nixpkgs applies patches which don't
    apply on the old version we're building..
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build!: update kube and related crates
    
    Brings back kube-proxy crate to this repo, for easier managing of code
    changes related to the openapi service.
    Updates kube to 0.94.2 and k8s-openapi to 0.22.0 with K8s v1_24
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: upgrade tracing, opentel, tonic, prost, rustls, tls and hyper
    
    Upgrade all aforementioned crates and some related crates.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: update most crate dependencies
    
    Updated all except some which are a bit more problematic:
    hyper
    rustls
    k8s-openapi
    kube
    hyper-rustls
    hyper-tls
    opentelemetry-otlp
    h2
    http
    actix-web
    actix-web-opentelemetry
    opentelemetry_sdk
    tracing-opentelemetry
    opentelemetry
    opentelemetry-semantic-conventions
    opentelemetry-http
    opentelemetry-jaeger
    http-body
    tower
    tower-http
    rustls-pemfile
    tonic
    prost
    tonic-build
    prost-build
    prost-derive
    prost-types
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    test: temporarily use previous openapi-generator-cli
    
    The newer versions have removed the python-prior generator and so
    upgrading is a considerable amount of work.
    Let's pin the nixpkgs with the openeapi-generator-cli and we'll
    tackle this as a separate task.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build: update nixpkgs and deps
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

File: control-plane/agents/Cargo.toml [hunk 1]
```diff
@@ -22,46 +21,47 @@ name = "jsongrpc"
...
-futures-util = { version = "0.3.28" }
-crossbeam-queue = "0.3.8"
-tower = { version = "0.4.13", features = [ "timeout", "util" ] }
-hyper = { version = "0.14.27", features = [ "client", "http1", "http2", "tcp", "stream" ] }
-opentelemetry = { version = "0.22.0" }
-tracing = "0.1.37"
-nix = { version = "0.27.1", default-features = false }
-prost-types = "0.12.1"
-url = "2.4.1"
+http = "1.1.0"
```
Comment (niladrih) [inline control-plane/agents/Cargo.toml]: nit: sort in alphabetical order? :)

File: control-plane/agents/src/bin/core/tests/watch/mod.rs [hunk 2]
```diff
@@ -37,17 +33,28 @@ async fn setup_watch(client: &dyn VolumeOperations) -> (Volume, tokio::sync::mps
...
+        loop {
+            let (stream, _) = listener.accept().await.unwrap();
+            let io = TokioIo::new(stream);
+
+            tokio::task::spawn(async move {
+                if let Err(err) = http1::Builder::new()
+                    .serve_connection(io, service_fn(notify))
+                    .await
+                {
+                    println!("Error serving connection: {:?}", err);
```
Comment (Abhinandan-Purkait) [inline control-plane/agents/src/bin/core/tests/watch/mod.rs]: println --> info ?

File: control-plane/grpc/src/tracing.rs [hunk 3]
```diff
@@ -239,7 +237,11 @@ fn update_span_from_response<T>(context: opentelemetry::Context, response: &Resp
         }
         None => tonic::Code::Ok,
     };
+    const RPC_GRPC_STATUS_CODE: &str = "rpc.grpc.status_code";
```
Comment (Abhinandan-Purkait) [inline control-plane/grpc/src/tracing.rs:240]: Why do we need to define it now? Is it removed from the previous imported path?

File: k8s/proxy/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+utils = { path = "../../utils/utils-lib" }
+
+kube = { version = "0.94.2", features = ["derive"] }
+tower = { version = "0.5.1", features = ["timeout", "util"] }
+tonic = "0.12.3"
+hyper-util = { version = "0.1.10", features = ["client", "client-legacy", "http1", "http2", "tokio", "service"] }
+hyper = { version = "1.5.0", features = ["client", "http1", "http2"] }
+hyper-body = { path = "../../utils/hyper-body" }
+
+anyhow = "1.0.92"
```
Comment (Abhinandan-Purkait) [inline k8s/proxy/Cargo.toml]: blank line

File: openapi/templates/default/tower-hyper/examples/example.rs
Comment (niladrih) [inline openapi/templates/default/tower-hyper/examples/example.rs:1]: 👍

**Review by niladrih — APPROVED** [review]

File: utils/hyper-body/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[package]
+name = "hyper-body"
+version = "0.1.0"
```
Comment (Abhinandan-Purkait) [inline utils/hyper-body/Cargo.toml:3]: Where do we use this new package?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: control-plane/agents/Cargo.toml (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/agents/Cargo.toml]: Moved it, but not everything is alphabetical, I'll leave that for another PR

File: control-plane/agents/src/bin/core/tests/watch/mod.rs (hunk 2 above)
Comment (tiagolobocastro) [reply control-plane/agents/src/bin/core/tests/watch/mod.rs]: Done, used error

File: control-plane/grpc/src/tracing.rs (hunk 3 above)
Comment (tiagolobocastro) [reply control-plane/grpc/src/tracing.rs:240]: Yes it was removed, it's now under some unsafe feature export, so I rather just hardcode it

File: utils/hyper-body/Cargo.toml (hunk 5 above)
Comment (tiagolobocastro) [reply utils/hyper-body/Cargo.toml:3]: openapi, kube-forward, kube-proxy and supportability

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**